pkg/start/start.go

```diff
@@ -229,9 +229,9 @@ func (o *Options) run(ctx context.Context, controllerCtx *Context, lock *resourc
 					time.Sleep(100 * time.Millisecond)
 					// if we still hold the leader lease, clear the owner identity (other lease watchers
 					// still have to wait for expiration) like the new ReleaseOnCancel code will do.
-					if err := lock.Update(resourcelock.LeaderElectionRecord{}); err == nil {
+					if err := lock.Update(localCtx, resourcelock.LeaderElectionRecord{}); err == nil {
 						// if we successfully clear the owner identity, we can safely delete the record
-						if err := lock.Client.ConfigMaps(lock.ConfigMapMeta.Namespace).Delete(lock.ConfigMapMeta.Name, nil); err != nil {
+						if err := lock.Client.ConfigMaps(lock.ConfigMapMeta.Namespace).Delete(ctx, lock.ConfigMapMeta.Name, metav1.DeleteOptions{}); err != nil {
```

Comment:
I understand the use of `localCtx` but why then the use of the parent `ctx`?  Assuming there's a good reason, and given there are so many Context variables in scope, a comment would be helpful.

Reply:
> I understand the use of `localCtx` but why then the use of the parent `ctx`?

Fixed to be `localCtx` with 23ba2bda -> 27b0531f.